src/circleci-base/scripts/docker-login.sh

```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+set -eo pipefail
```

Comment:
Either add the `set -u` flag here please, or perform explicit variable checks, to ensure the global variables exist and are not empty.